src/test/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitCheckerTest.java

```diff
@@ -0,0 +1,301 @@
+                            if ("application/json"
+                                    .equals(response.getHeaders().getContentTypeHeader().mimeTypePart())) {
+                                return Response.Builder.like(response)
+                                        .but()
+                                        .body(response.getBodyAsString()
+                                                .replace("https://api.github.com/",
+                                                        "http://localhost:" + githubApi.port() + "/")
+                                                .replace("https://raw.githubusercontent.com/",
+                                                        "http://localhost:" + githubRaw.port() + "/")
+                                        )
+                                        .build();
+                            }
+                            return response;
+                        }
+
+                        @Override
+                        public String getName() {
+                            return "url-rewrite";
+                        }
+
+                    })
+    );
+    private GitHub github;
+
+    private final ApiRateLimitChecker apiRateLimitCheckerThrottleOnOver = ThrottleOnOver;
+    private final ApiRateLimitChecker apiRateLimitCheckerThrottleForNormalize = ApiRateLimitChecker.ThrottleForNormalize;
+    private final Random entropy = new Random(1000);
+
+    @Before
+    public void setUp() throws Exception {
+        new File("src/test/resources/api/mappings").mkdirs();
+        new File("src/test/resources/api/__files").mkdirs();
+        new File("src/test/resources/raw/mappings").mkdirs();
+        new File("src/test/resources/raw/__files").mkdirs();
+        githubApi.enableRecordMappings(new SingleRootFileSource("src/test/resources/api/mappings"),
+                new SingleRootFileSource("src/test/resources/api/__files"));
+        githubRaw.enableRecordMappings(new SingleRootFileSource("src/test/resources/raw/mappings"),
+                new SingleRootFileSource("src/test/resources/raw/__files"));
+
+
+        githubApi.stubFor(
+                get(urlMatching(".*")).atPriority(10).willReturn(aResponse().proxiedFrom("https://api.github.com/")));
+        githubRaw.stubFor(get(urlMatching(".*")).atPriority(10)
+                .willReturn(aResponse().proxiedFrom("https://raw.githubusercontent.com/")));
+
+        github = Connector.connect("http://localhost:" + githubApi.port(), null);
+
+        resetAllScenarios();
+
+        handler = new RingBufferLogHandler(100);
+        final Logger logger = Logger.getLogger(getClass().getName());
+        logger.addHandler(handler);
+        listener = new LogTaskListener(logger, Level.INFO);
+    }
+
+    private void setupStubs(List<JsonRateLimit> scenarios) {
+        githubApi.listAllStubMappings().getMappings().stream()
+                .filter(x -> x.getScenarioName() != null && x.getScenarioName() == ("API Rate Limit"))
```

Comment:
Since you're generating the mappings and stubs, instead of having to delete all the mappings before each run, how about using a random scenario name?